iothub/service/src/RetryPolicies/IotHubServiceExponentialBackoffRetryPolicy.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Represents a retry policy that performs a specified number of retries, using a exponential back-off scheme, with option jitter,
+    /// to determine the interval between retries.
+    /// </summary>
+    /// <remarks>
+    /// Jitter can change the delay from 95% to 105% of the calculated time.
+    /// </remarks>
+    public class IotHubServiceExponentialBackoffRetryPolicy : IotHubServiceRetryPolicyBase
+    {
+        // If we start with an exponent of 1 to calculate the number of millisecond delay, it starts too low and takes too long to get over 1 second.
+        // So we always add 6 to the retry count to start at 2^7=128 milliseconds, and exceed 1 second delay on retry #4.
+        private const uint MinExponent = 6u;
+
+        // Avoid integer overlow (max of 32) and clamp max delay.
+        private const uint MaxExponent = 32u;
+
+        private readonly TimeSpan _maxDelay;
+        private readonly bool _useJitter;
+
+        /// <summary>
+        /// Creates an instance of this class.
+        /// </summary>
+        /// <param name="maxRetries">The maximum number of retry attempts; use 0 for infinite retries.</param>
+        /// <param name="maxWait">The maximum amount of time to wait between retries (will not exceed ~12.43 days).</param>
+        /// <param name="useJitter">Whether to add a small, random adjustment to the retry delay to avoid synchronicity in clients retrying.</param>
+        /// <exception cref="ArgumentOutOfRangeException">Throw if the value of <paramref name="maxWait"/> is negative.</exception>
+        public IotHubServiceExponentialBackoffRetryPolicy(uint maxRetries, TimeSpan maxWait, bool useJitter = true)
+            : base(maxRetries)
+        {
+            if (maxWait.Ticks < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(maxWait), maxWait.Ticks, string.Format(CultureInfo.CurrentCulture, "ArgumentCannotBeNegative", new object[]
```

Comment:
Why not do what we had in the client one? `Argument.AssertNotNegativeValue(maxWait.Ticks, nameof(maxWait));`